test/services/external_policy_remover_test.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+# A class to test removal of external policies
+class ExternalPolicyRemoverTest < ActiveSupport::TestCase
+  test 'does nothing if no external policies exist' do
+    assert_empty Profile.external.where(policy_id: nil)
+    assert_difference('Profile.count' => 0) do
+      ExternalPolicyRemover.run!
+    end
+  end
+
+  test 'removes all external policies' do
+    profiles(:one).dup.update!(external: true, account: accounts(:test))
+    profiles(:two).dup.update!(external: true, policy_object: policies(:one),
+                               account: accounts(:test))
+    assert_equal 1, Profile.external.where(policy_id: nil).count
+    assert_difference('Profile.count' => -1) do
+      ExternalPolicyRemover.run!
```

Comment:
Suggestion, to add test results and rules to those profiles and check whether they are deleted. It could uncover a regression, as there is a luck of foreign keys.

Reply:
IMO this case should be covered in the profile model tests; I'll ensure they exist there.